### ISSUE TITLE
Implement a few missing features in Kuma's support for Envoy access log format

### ISSUE THREAD

**Issue by yskopets** [kumahq/kuma#596]
### Summary

The initial implementation is missing the following features:
1. `%START_TIME%` commands ignore the user-defined format string
2. `%DYNAMIC_METADATA(NAMESPACE:KEY*):Z%` commands return a stub value
3. `%FILTER_STATE(KEY):Z%` commands return a stub value

**Comment by lahabana** [comment]
The biggest ask atm has been to support environment variables in access log format.

**Comment by lahabana** [comment]
But let's have a look at: https://www.envoyproxy.io/docs/envoy/latest/configuration/observability/access_log/usage and do an inventory on everything we don't currently support and look at 1) why 2) how easy would it be to add them

**Comment by lahabana** [comment]
People are mostly asking about `%ENVIRONMENT(X):Z%` support there's a core problem that if using GRPC access logging the logging format is reprocessed in the DP which is different than the application container. Seems like you'd be able to achieve this with Custom metadata (with [custom_tags](https://www.envoyproxy.io/docs/envoy/latest/api-v3/extensions/access_loggers/grpc/v3/als.proto#extensions-access-loggers-grpc-v3-commongrpcaccesslogconfig)) it's not something we have because we're using older protos but go-control-plane has more recent ones: https://github.com/envoyproxy/go-control-plane/blob/ae885253afafeeb9a24fdc29b9b1e9950a105430/envoy/extensions/access_loggers/grpc/v3/als.pb.go#L163

To summarise I think we should expand the current setup to support environment variables using custom tags (probably need to update go-control-plane to do this) and shelf the rest of the support.

For other things we'll need to reconsider this when we roll out the new selectors and rewrite the access log policy as what we do is overly complicated and has poor UX.
Does this make sense?
